### PR TITLE
Init implementation of serverless function for processing ESRI API url

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Init implementation of serverless function for processing ESRI API url

Test cases are included

### 15th May 2020
Update logic due to ticket https://github.com/magda-io/magda/issues/2830

### Checklist

-   [x] There are unit tests to verify my changes are correct

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,3 +1,101 @@
-export default async function myFunction(input: any) {
-    return "Hello world!\n";
+import axios from "axios";
```
Comment (AlexGilleran) [inline src/index.ts:1]: Can we stick to `fetch` ?

File: src/index.ts [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+            data.extent.xmin,
+            data.extent.ymin,
+            data.extent.xmax,
+            data.extent.ymax
+        ],
+        spatialDataInputMethod: "bbox"
+    };
+}
+
+export default async function myFunction(
```
Comment (AlexGilleran) [inline src/index.ts:34]: Maybe a better name than `myFunction`? ;)

File: src/index.ts [hunk 3]
```diff
@@ -1,3 +1,101 @@
...
+            aspects: {
+                "dcat-dataset-strings": {
+                    title: data.name,
+                    description: data?.description
+                }
+            }
+        },
+        distributions: [
+            {
+                id: data.id,
```
Comment (AlexGilleran) [inline src/index.ts:78]: Should we be prefixing this to make sure it doesn't collide? Maybe with the URL of the source?

File: src/index.ts (hunk 1 above)
Comment (t83714) [reply src/index.ts:1]: 👍 will use `isomorphic-fetch` instead~

File: src/index.ts (hunk 2 above)
Comment (t83714) [reply src/index.ts:34]: will rename it to `processEsriApiUrl` --- if you have a better name, just let me know

File: src/index.ts (hunk 3 above)
Comment (t83714) [reply src/index.ts:78]: Do you mean the `id`? Here try to just return the origin id. It actually isn't used at front-end --- we still use the `magda-dist-xx` id at frontend (to make sure there isn't frontend code breaks).
Do we have to use the original id? Or it doesn't matter?

**Comment by t83714** [comment]
@AlexGilleran removed `axios` and changed the function name.
Please have another look when you have time~

**Pull request merged**